adagrams/game.py

```diff
@@ -1,13 +1,166 @@
 from random import randint
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+# wave_01:
+# we need to create function that select 10 letters randomly.
+# make sure to return letters as list of strings.
+# draw_letter() letter selected no  ore times that its available.
+# test3 was not passing so I looked up chatgt to see why, went over it with Mikelle to understand.
 
 def draw_letters():
-    pass
+    letter_list = [] # store letters based on their frequancy
+    for letter, count in LETTER_POOL.items(): # we used .items() because we need to return a dict with letters:key, count:value
+        for i in range(count):
+            letter_list.append(letter)
+    print(letter_list)
+
+
+    pulled_letters = [] # pull random letters
+    for i in range(10): # max 10 letters
+        random_index = randint(0, len(letter_list) - 1) # randint generates random int between 2 values.
+        random_letter = letter_list.pop(random_index) # used pop method with help of Mikelle in office hours :)
```

Comment:
Any function that removes an item from the middle of the list is going to have to shift every other item down in order to fill the space that was made. This increases the complexity from `O(1)` to `O(n)` which isn't terrible, but it is an increase. There are a couple of ways that we could approach this, but my favorite is to start by finding the random index. Once you've done that, you can swap the value at the index with the value at the end of the list and then use the indexless `pop()` method. While this takes an extra step, it does decrease the complexity. Overall, this approach is a really nice approach when it comes to removing a random value from a list!